modules/extension/wmts/src/main/java/org/geotools/ows/wmts/WebMapTileServer.java

```diff
@@ -245,22 +244,13 @@ public GetTileResponse issueRequest(GetSingleTileRequest tileRequest)
 
     /** Creates a GetMultiTileRequest. */
     public GetTileRequest createGetTileRequest() {
-
-        OperationType getTile = getCapabilities().getRequest().getGetTile();
-        URL url = findGetTileURL(getTile);
-
-        if (url == null) {
-            type = WMTSServiceType.REST;
-            url = getCapabilities().getService().getOnlineResource();
-        }
-
         Properties props = new Properties();
         props.put("type", type.name());
 
         GetTileRequest request =
                 getSpecification()
                         .createGetMultiTileRequest(
-                                url, props, getCapabilities(), this.getHTTPClient());
+                                serverURL, props, getCapabilities(), this.getHTTPClient());
```

Comment:
I'm not sure about this change... the WMTS specification allows a server to have the capability URL on one host/port/path and the GetTile on an entirely different combination (yes, on a different host)... and other calls or RESTful endpoints can also be spread at different locations. 

A compliant client should always use the values provided in the operations section of the Capabilities document.

Reply:
Ok, it was an easy way to pass on extra parameter's, and it will only work for KVP.
@aaime What if I check that the operations capabilities are respected?

Reply:
What do you mean?

Reply:
What I was thinking of was a function that checked if the serverURL have the same host / path as the one specified within the operations section of the capabilities.

Reply:
Ok but... by specification you should not really use the server URL (there is no such a thing, it's just the capabilities URL), once you have the capabilities document, all you need to follow are either the backlinks in the Operations section (if KVP) or the Resource URLs (if RESTful).

Reply:
Looking at the code seems to confirm my suspicion, the URL provided in the constructor will be used as the base URL to build the final GetTile URL (becomes "onlineResource" in the request classes). As said, this breaks compliance, it would make the changed code work only against WMTS server that co-locate all services on the same machine and path, while the spec allows freedom to put single request on different hosts and path, and in the case of REST requests, to have specific endpoints for different layers (e.g., a bit dynamic, a bit static on S3, and so on).